lib/util/http/gitea.ts

```diff
@@ -37,7 +37,7 @@ export class GiteaHttp extends Http<GiteaHttpOptions, GiteaHttpOptions> {
     path: string,
     options?: InternalHttpOptions & GiteaHttpOptions
   ): Promise<HttpResponse<T> | null> {
-    const resolvedUrl = resolveUrl(path);
+    const resolvedUrl = resolveUrl(path, options.baseUrl ?? baseUrl);
```

Comment:
Change is required to pass and use new `baseUrl` which can be different from platform `baseUrl`